### PR TITLE
Update sql-faq.md

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

closes https://github.com/pingcap/tidb/issues/19818

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3851 https://github.com/pingcap/docs/pull/3864
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Have version specific changes 
- [x] Might cause conflicts

### CODE REVIEW

File: faq/sql-faq.md [hunk 1]
```diff
@@ -12,6 +12,66 @@ aliases: ['/docs-cn/dev/faq/sql-faq/']
...
+| a    | b    |
++------+------+
+|    1 |    1 |
+|    2 |    2 |
++------+------+
+2 rows in set (0.00 sec)
+```
+
+```sql
+> select * from t; -- the order of results is not guaranteed
```
Comment (SunRunAway) [inline faq/sql-faq.md:39]: 这里的注释要不要也翻译一下？

**Comment by yikeke** [comment]
label: translation/doing -> translation/from-docs-cn? @TomShawn

File: faq/sql-faq.md [hunk 2]
```diff
@@ -12,6 +12,66 @@ aliases: ['/docs-cn/dev/faq/sql-faq/']
 
 详细可参考[系统变量](/system-variables.md)。
 
+## 省略 `ORDER BY` 时 TiDB 中返回结果的顺序与 MySQL 中的不一致
```
Comment (yikeke) [inline faq/sql-faq.md:15]:
```suggestion
## 省略 `ORDER BY` 条件时 TiDB 中返回结果的顺序与 MySQL 中的不一致

```

File: faq/sql-faq.md [hunk 3]
```diff
@@ -12,6 +12,66 @@ aliases: ['/docs-cn/dev/faq/sql-faq/']
...
++------+------+
+|    1 |    1 |
+|    2 |    1 |
+|    2 |    2 |
++------+------+
+3 rows in set (0.00 sec)
+```
+
+```sql
+> select * from t order by a; -- the order of column a is guaranteed, but b is not
```
Comment (yikeke) [inline faq/sql-faq.md:64]: to be translated

**Review by yikeke — COMMENTED** [review]
Rest LGTM

File: faq/sql-faq.md (hunk 3 above)
Comment (TomShawn) [reply faq/sql-faq.md]: done in https://github.com/pingcap/docs-cn/pull/4500/commits/f93aee243d1fd7282be1eb4c76795b8720edf1c9

File: faq/sql-faq.md (hunk 1 above)
Comment (TomShawn) [reply faq/sql-faq.md]: done in https://github.com/pingcap/docs-cn/pull/4500/commits/f93aee243d1fd7282be1eb4c76795b8720edf1c9

**Review by yikeke — APPROVED** [review]
LGTM

**Review by SunRunAway — COMMENTED** [review]
LGTM

**Pull request merged**